src/wifi.rs

```diff
@@ -194,22 +194,26 @@ impl TryFrom<&ClientConfiguration> for Newtype<wifi_sta_config_t> {
     }
 }
 
-impl From<Newtype<wifi_sta_config_t>> for ClientConfiguration {
+impl From<Newtype<wifi_sta_config_t>> for Option<ClientConfiguration> {
```

Comment:
Why don't you keep the original code as-is, and only change
```rust
auth_method: Newtype(conf.0.threshold.authmode).into()
```

to
```rust
auth_method: Newtype(conf.0.threshold.authmode).into().unwrap()
```

?

Reply:
Hm actually, yeah good point..., let me adjust stuff